Mathlib/RingTheory/MvPowerSeries/NoZeroDivisors.lean

```diff
@@ -65,6 +66,23 @@ theorem mem_nonZeroDivisors_of_constantCoeff {φ : MvPowerSeries σ R}
 
 end Semiring
 
+section
+
+variable {σ R : Type*} [Semiring R] [NoZeroDivisors R]
+
+instance : NoZeroDivisors (MvPowerSeries σ R) where
+  eq_zero_or_eq_zero_of_mul_eq_zero {φ ψ} h := by
+    letI : LinearOrder σ := LinearOrder.swap σ WellOrderingRel.isWellOrder.linearOrder
+    letI : WellFoundedGT σ := by
+      unfold WellFoundedGT
+      suffices IsWellFounded σ fun x y ↦ WellOrderingRel x y by
+        exact this
```

Comment:
Do you need these lines? Or is it enough to have
```suggestion
      change IsWellFounded σ fun x y ↦ WellOrderingRel x y
```

Reply:
K

Reply:
OK.